src/v/net/probes.cc

```diff
@@ -404,6 +404,9 @@ void tls_certificate_probe::loaded(
         return;
     }
 
+    _cert_expiry_time = clock_type::time_point::max();
+    _ca_expiry_time = clock_type::time_point::max();
```

Comment:
Should reset really be setting these to max when `!certs_info.has_value() || !ts_info.has_value()`?

Should that check also ensure they are not empty?

Reply:
Presumably they should be cleared down to something nominally invalid in either case, right? Since we're monitoring a file, I assumed whatever state we previously had in the probe should be 100% junk, irrespective of whether the new creds are valid, empty, etc. 

That said, it's quite possible/likely that I misunderstand the common case for administering TLS certificates.

Reply:
I would think that min would be a safer number than max; I'd probably have a check that the cert doesn't expire in the next week or so, and max would satisfy that. I'm assuming here that when the callback is called we are expecting valid certificates. But maybe it means that TLS certificates have been turned off?

Reply:
>have a check that the cert doesn't expire in the next week or so, and max would satisfy that

Not sure I follow. I was thinking about something like 
- always reset probe states to time_point::min (effectively "invalid")
- use a couple of temporary optionals to track the min expiry in the creds
- set probe state to the new value at the very end

Does that cover our bases? Is there an edge case I'm missing?

Reply:
Took sort of a hard left turn - trying to make it harder to represent invalid states. Very much open to suggestions here.